### PR TITLE
add CRIO support

### PR DESCRIPTION
**Pull request by jsalatiel**
Add CRIO runtime support.

### CODE REVIEW

File: create_crio_sysext.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+fi
+
+rm -f "cri-o.${ARCH}.v${VERSION}.tar.gz"
+curl -o "cri-o.${ARCH}.v${VERSION}.tar.gz" -fsSL "https://storage.googleapis.com/cri-o/artifacts/cri-o.${ARCH}.v${VERSION}.tar.gz" 
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}" "${SYSEXTNAME}/tmp"
+tar --force-local -xf cri-o.${ARCH}.v${VERSION}.tar.gz -C "${SYSEXTNAME}/tmp"
+cd ${SYSEXTNAME}/tmp/cri-o/
+sed -i '/^sed -i.*DESTDIR/d' install # removes sed replacements from install script to keep the default location (/usr) in the base config file
+DESTDIR="${SCRIPTFOLDER}/${SYSEXTNAME}" PREFIX=/usr ETCDIR=$PREFIX/share/crio/etc OCIDIR=$PREFIX/share/oci-umount/oci-umount.d \
```
Comment (pothos) [inline create_crio_sysext.sh:36]:
```suggestion
DESTDIR="${PWD}/${SYSEXTNAME}" PREFIX=/usr ETCDIR=$PREFIX/share/crio/etc OCIDIR=$PREFIX/share/oci-umount/oci-umount.d \
```

File: create_crio_sysext.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+export ARCH="${ARCH-x86-64}"
+SCRIPTFOLDER="$(dirname "$(readlink -f "$0")")"
+
+if [ $# -lt 2 ] || [ "$1" = "-h" ] || [ "$1" = "--help" ]; then
+  echo "Usage: $0 VERSION SYSEXTNAME"
+  echo "The script will download the cri-o release binaries (e.g., for v1.28.4) and create a sysext squashfs image with the name SYSEXTNAME.raw in the current folder."
```
Comment (pothos) [inline create_crio_sysext.sh:9]:
```suggestion
  echo "The script will download the cri-o release binaries (e.g., for 1.28.4) and create a sysext squashfs image with the name SYSEXTNAME.raw in the current folder."
```

File: create_crio_sysext.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+rm -f "cri-o.${ARCH}.v${VERSION}.tar.gz"
+curl -o "cri-o.${ARCH}.v${VERSION}.tar.gz" -fsSL "https://storage.googleapis.com/cri-o/artifacts/cri-o.${ARCH}.v${VERSION}.tar.gz" 
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}" "${SYSEXTNAME}/tmp"
+tar --force-local -xf cri-o.${ARCH}.v${VERSION}.tar.gz -C "${SYSEXTNAME}/tmp"
+cd ${SYSEXTNAME}/tmp/cri-o/
+sed -i '/^sed -i.*DESTDIR/d' install # removes sed replacements from install script to keep the default location (/usr) in the base config file
+DESTDIR="${PWD}/${SYSEXTNAME}" PREFIX=/usr ETCDIR=$PREFIX/share/crio/etc OCIDIR=$PREFIX/share/oci-umount/oci-umount.d \
+  CNIDIR=$PREFIX/share/crio/cni/etc/net.d/ OPT_CNI_BIN_DIR=$PREFIX/share/crio/cni/bin/  BASHINSTALLDIR=/tmp FISHINSTALLDIR=/tmp ZSHINSTALLDIR=/tmp MANDIR=/tmp ./install 
+cd "${SCRIPTFOLDER}"
```
Comment (pothos) [inline create_crio_sysext.sh:38]:
```suggestion
cd -
```
This switches back to the previous folder

File: create_crio_sysext.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+fi
+
+rm -f "cri-o.${ARCH}.v${VERSION}.tar.gz"
+curl -o "cri-o.${ARCH}.v${VERSION}.tar.gz" -fsSL "https://storage.googleapis.com/cri-o/artifacts/cri-o.${ARCH}.v${VERSION}.tar.gz" 
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}" "${SYSEXTNAME}/tmp"
+tar --force-local -xf cri-o.${ARCH}.v${VERSION}.tar.gz -C "${SYSEXTNAME}/tmp"
+cd ${SYSEXTNAME}/tmp/cri-o/
+sed -i '/^sed -i.*DESTDIR/d' install # removes sed replacements from install script to keep the default location (/usr) in the base config file
+DESTDIR="${PWD}/${SYSEXTNAME}" PREFIX=/usr ETCDIR=$PREFIX/share/crio/etc OCIDIR=$PREFIX/share/oci-umount/oci-umount.d \
```
Comment (pothos) [inline create_crio_sysext.sh:36]:
```suggestion
DESTDIR="${PWD}/../../${SYSEXTNAME}" PREFIX=/usr ETCDIR=$PREFIX/share/crio/etc OCIDIR=$PREFIX/share/oci-umount/oci-umount.d \
```

File: create_crio_sysext.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+fi
+
+rm -f "cri-o.${ARCH}.v${VERSION}.tar.gz"
+curl -o "cri-o.${ARCH}.v${VERSION}.tar.gz" -fsSL "https://storage.googleapis.com/cri-o/artifacts/cri-o.${ARCH}.v${VERSION}.tar.gz" 
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}" "${SYSEXTNAME}/tmp"
+tar --force-local -xf cri-o.${ARCH}.v${VERSION}.tar.gz -C "${SYSEXTNAME}/tmp"
+cd ${SYSEXTNAME}/tmp/cri-o/
+sed -i '/^sed -i.*DESTDIR/d' install # removes sed replacements from install script to keep the default location (/usr) in the base config file
+DESTDIR="${PWD}/../../${SYSEXTNAME}" PREFIX=/usr ETCDIR=$PREFIX/share/crio/etc OCIDIR=$PREFIX/share/oci-umount/oci-umount.d \
```
Comment (pothos) [inline create_crio_sysext.sh:36]:
```suggestion
DESTDIR="${PWD}/../../../${SYSEXTNAME}" PREFIX=/usr ETCDIR=$PREFIX/share/crio/etc OCIDIR=$PREFIX/share/oci-umount/oci-umount.d \
```

File: create_crio_sysext.sh [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  ARCH="amd64"
+elif [ "${ARCH}" = "aarch64" ]; then
+  ARCH="arm64"
+fi
+
+rm -f "cri-o.${ARCH}.v${VERSION}.tar.gz"
+curl -o "cri-o.${ARCH}.v${VERSION}.tar.gz" -fsSL "https://storage.googleapis.com/cri-o/artifacts/cri-o.${ARCH}.v${VERSION}.tar.gz" 
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}" "${SYSEXTNAME}/tmp"
+tar --force-local -xf cri-o.${ARCH}.v${VERSION}.tar.gz -C "${SYSEXTNAME}/tmp"
```
Comment (pothos) [inline create_crio_sysext.sh:33]:
```suggestion
tar --force-local -xf "cri-o.${ARCH}.v${VERSION}.tar.gz" -C "${SYSEXTNAME}/tmp"
```

File: create_crio_sysext.sh [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+elif [ "${ARCH}" = "aarch64" ]; then
+  ARCH="arm64"
+fi
+
+rm -f "cri-o.${ARCH}.v${VERSION}.tar.gz"
+curl -o "cri-o.${ARCH}.v${VERSION}.tar.gz" -fsSL "https://storage.googleapis.com/cri-o/artifacts/cri-o.${ARCH}.v${VERSION}.tar.gz" 
+rm -rf "${SYSEXTNAME}"
+mkdir -p "${SYSEXTNAME}" "${SYSEXTNAME}/tmp"
+tar --force-local -xf "cri-o.${ARCH}.v${VERSION}.tar.gz" -C "${SYSEXTNAME}/tmp"
+cd ${SYSEXTNAME}/tmp/cri-o/
```
Comment (pothos) [inline create_crio_sysext.sh:34]:
```suggestion
cd "${SYSEXTNAME}/tmp/cri-o/"
```

**Review by pothos — APPROVED** [review]
I had to carry over some of the suggestions of the last PR and directly applied them here

**Comment by jsalatiel** [comment]
Thank-you

**Pull request merged**